ui/Dockerfile

```diff
@@ -4,6 +4,7 @@ RUN mkdir /home/node/ui
 WORKDIR /home/node/ui
```

Comment:
@coderabbitai, this docker file is giving below error while building, can you please help me to fix the issue

```
Error: EACCES: permission denied, open '/home/node/ui/.yarn/install-state.gz'
```

Reply:
@coderabbitai, Still receiving the same error message after change, please help me to fix it.

Reply:
@coderabbitai, now the error is 

```
 chown: changing ownership of '/home/node/ui/Dockerfile': Operation not permitted
```

Reply:
@coderabbitai, now the error is as below

```
1.608 ➤ YN0000: ┌ Link step
11.38 ➤ YN0001: │ Error: EACCES: permission denied, open '/home/node/ui/.yarn/install-state.gz'
11.38 ➤ YN0000: └ Completed in 9s 768ms
```

Reply:
@coderabbitai, now the error is as below inspite of adjusting the permissions

```
1.579 ➤ YN0000: ┌ Link step
14.08 ➤ YN0001: │ Error: EACCES: permission denied, open '/home/node/ui/.yarn/install-state.gz'
14.08 ➤ YN0000: └ Completed in 12s 500ms
14.08 ➤ YN0000: Failed with errors in 13s 267ms
------
failed to solve: process "/bin/sh -c mkdir -p /home/node/ui/node_modules &&     yarn install &&     chmod -R 755 /home/node/ui/.yarn  # Ensure correct permissions for .yarn directory" did not complete successfully: exit code: 1
```

Reply:
@coderabbitai, still the same below issue after adjust script, i'm  building image from docker-compose.yml where script is as follows:

```shell
  ui:
    build: ./ui
    user: node
    volumes:
      - ./ui:/home/node/ui
      - folioman-ui-node-modules:/home/node/ui/node_modules
    working_dir: /home/node/ui
    command: sh -c "yarn install && npm run build && npm run start"
```

```
11.07 ➤ YN0001: │ Error: EACCES: permission denied, open '/home/node/ui/.yarn/install-state.gz'
11.07 ➤ YN0000: └ Completed in 9s 571ms
11.07 ➤ YN0000: Failed with errors in 10s 289ms
------
failed to solve: process "/bin/sh -c mkdir -p /home/node/ui/node_modules &&     yarn install &&     chmod -R 755 /home/node/ui/.yarn &&     npm run build" did not complete successfully: exit code: 1
```

Reply:
@coderabbitai, I have followed all the steps still receiving below error message

```
1.508 ➤ YN0000: ┌ Link step
10.78 ➤ YN0001: │ Error: EACCES: permission denied, open '/home/node/ui/.yarn/install-state.gz'
10.78 ➤ YN0000: └ Completed in 9s 274ms
10.78 ➤ YN0000: Failed with errors in 10s 8ms
------
failed to solve: process "/bin/sh -c mkdir -p /home/node/ui/node_modules &&     yarn install &&     chmod -R 755 /home/node/ui/.yarn &&     npm run build" did not complete successfully: exit code: 1
```

Reply:
@coderabbitai, now I have another new error, I have resolved .yarn/install-state.gz issue

```
RUN mkdir -p /home/node/ui/node_modules &&     yarn install &&     chmod -R 755 /home/node/ui/.yarn  &&     npm run build:
1.911 Internal Error: EACCES: permission denied, mkdir '/home/node/ui/.yarn/cache'
1.911 Error: EACCES: permission denied, mkdir '/home/node/ui/.yarn/cache'
------
failed to solve: process "/bin/sh -c mkdir -p /home/node/ui/node_modules &&     yarn install &&     chmod -R 755 /home/node/ui/.yarn  &&     npm run build" did not complete successfully: exit code: 1
```

Reply:
@coderabbitai, still same issue. my current docker-compose file is 

```shell
  ui:
    build: ./ui
    user: node
    volumes:
      - ./ui:/home/node/ui
      - folioman-ui-node-modules:/home/node/ui/node_modules
    working_dir: /home/node/ui
    command: sh -c "yarn install && npm run build && npm run start"
``` 

and current Dockerfile is 

```
FROM node:16-buster
# Set default user and group ID for all operations
RUN mkdir -p /home/node/ui && \
    chown -R node:node /home/node/ui && \
    chmod -R 755 /home/node/ui

# Switch to user 'node' for remaining operations
USER node
WORKDIR /home/node/ui
COPY . .
RUN mkdir -p /home/node/ui/node_modules && \
    yarn install && \
    chmod -R 755 /home/node/ui/.yarn && \
    # Ensure correct permissions for .yarn directory and any subdirectories
    npm run build
CMD npm run start
```

Reply:
@coderabbitai, followed the recommended steps, still facing the same issue as below.

```
 > [ui 5/5] RUN mkdir -p /home/node/ui/node_modules &&     yarn install &&     chmod -R 755 /home/node/ui/.yarn &&     npm run build:
1.955 Internal Error: EACCES: permission denied, mkdir '/home/node/ui/.yarn/cache'
1.955 Error: EACCES: permission denied, mkdir '/home/node/ui/.yarn/cache'
------
failed to solve: process "/bin/sh -c mkdir -p /home/node/ui/node_modules &&     yarn install &&     chmod -R 755 /home/node/ui/.yarn &&     npm run build" did not complete successfully: exit code: 1
```

Reply:
@coderabbitai, now receiving below error, below 3 files are present in my ui folder already

```
FROM node:16-buster
# Set default user and group ID for all operations
RUN mkdir -p /home/node/ui && \
    chown -R node:node /home/node/ui && \
    chmod -R 755 /home/node/ui

# Switch to user 'node' for remaining operations
USER node
WORKDIR /home/node/ui
COPY . .
RUN mkdir -p /home/node/ui/node_modules /home/node/ui/.yarn && \
    chmod -R 755 /home/node/ui/.yarn && \
    yarn install && \
    npm run build
CMD npm run start
```

Reply:
@coderabbitai, updated the error in above comment, its my mistake in copy pasting.

Reply:
@coderabbitai, now back to starting issue. when `yarn install` is executed it is creating `cache` folder  and .yarn folder is already mounted from ui folder. Hence we have mismatch

now receving below error
```
[ui 5/5] RUN mkdir -p /home/node/ui/node_modules &&     yarn install &&     npm run build:
1.963 Internal Error: EACCES: permission denied, mkdir '/home/node/ui/.yarn/cache'
1.963 Error: EACCES: permission denied, mkdir '/home/node/ui/.yarn/cache'
------
failed to solve: process "/bin/sh -c mkdir -p /home/node/ui/node_modules &&     yarn install &&     npm run build" did not complete successfully: exit code: 1
```

Reply:
@coderabbitai, when I excluded .yarn folder then below is the error, so .yarn folder which is under ui should be mounted and only then yarn install can be executed.

```
 => CACHED [ui 3/5] WORKDIR /home/node/ui                                                                     0.0s
 => [ui 4/5] COPY . .                                                                                         0.0s
 => ERROR [ui 5/5] RUN mkdir -p /home/node/ui/node_modules &&     yarn install &&     npm run build           1.6s
------
 > [ui 5/5] RUN mkdir -p /home/node/ui/node_modules &&     yarn install &&     npm run build:
1.532 node:internal/modules/cjs/loader:1031
1.532   throw err;
1.532   ^
1.532 
1.532 Error: Cannot find module '/home/node/ui/.yarn/releases/yarn-3.2.2.cjs'
1.532     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1028:15)
1.532     at Function.Module._load (node:internal/modules/cjs/loader:873:27)
1.532     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
1.532     at node:internal/main/run_main_module:22:47 {
1.532   code: 'MODULE_NOT_FOUND',
1.532   requireStack: []
1.532 }
------
failed to solve: process "/bin/sh -c mkdir -p /home/node/ui/node_modules &&     yarn install &&     npm run build" did not complete successfully: exit code: 1
```

Reply:
@coderabbitai, below is the erroe message now

```
 => ERROR [ui 5/5] RUN mkdir -p /home/node/ui/node_modules &&     yarn install &&     npm ru  1.9s
------
 > [ui 5/5] RUN mkdir -p /home/node/ui/node_modules &&     yarn install &&     npm run build:
1.857 Internal Error: EACCES: permission denied, mkdir '/home/node/ui/.yarn/cache'
1.857 Error: EACCES: permission denied, mkdir '/home/node/ui/.yarn/cache'
------
failed to solve: process "/bin/sh -c mkdir -p /home/node/ui/node_modules &&     yarn install &&     npm run build" did not complete successfully: exit code: 1
```

Reply:
@coderabbitai, above is not working because to run `yarn install` we need the files mounted from `ui/.yarn` folder